### ISSUE TITLE
Would same model perform differently on one signle GPU or multiple GPUs

### ISSUE THREAD

**Issue by CHELSEA234** [tensorflow/serving#508]
I tried to implement a model working on 2 Gpus, but the result is different with when I tried same model with identical dataset on one single GPU.

Here is what I did on two GPUs code:
`with tf.device('/gpu:0'):
    conv1_0 = tf.nn.relu(tf.nn.conv2d(images_0, weights['w1'], strides=[1,1,1,1], padding='VALID'))
    conv2_0 = tf.nn.relu(tf.nn.conv2d(conv1_0, weights['w2'], strides=[1,1,1,1], padding='VALID'))
    pred_0 = tf.nn.conv2d(conv2_0, weights['w3'], strides=[1,1,1,1], padding='VALID')
    loss_0 = tf.reduce_mean(tf.square(labels_0 - pred_0))
    total_loss.append(loss_0)

with tf.device('/gpu:1'):
    conv1_1 = tf.nn.relu(tf.nn.conv2d(images_1, weights['w1'], strides=[1,1,1,1], padding='VALID'))
    conv2_1 = tf.nn.relu(tf.nn.conv2d(conv1_1, weights['w2'], strides=[1,1,1,1], padding='VALID'))
    pred_1 = tf.nn.conv2d(conv2_1, weights['w3'], strides=[1,1,1,1], padding='VALID')
    loss_1 = tf.reduce_mean(tf.square(labels_1 - pred_1))
    total_loss.append(loss_1)

with tf.device('/cpu:0'):
    # total_loss = (loss_1+loss_2)/2.0
    total_loss_sum = tf.add_n(total_loss)/2.0
    solver = tf.train.AdamOptimizer(0.0001).minimize(total_loss_sum)
    conv1 = tf.nn.relu(tf.nn.conv2d(images, weights['w1'], strides=[1,1,1,1], padding='VALID'))
    conv2 = tf.nn.relu(tf.nn.conv2d(conv1, weights['w2'], strides=[1,1,1,1], padding='VALID'))
    pred = tf.nn.conv2d(conv2, weights['w3'], strides=[1,1,1,1], padding='VALID')
#######################training phase#######################
training_dataset, label_dataset, testing_dataset, testing_num = DL.data_loading('Yang91','Set5')
training_dataset_number = training_dataset.shape[0]
training_dataset_0 = training_dataset[:int(training_dataset_number/2.0),:,:,:]
training_dataset_1 = training_dataset[int(training_dataset_number/2.0)+1:int(training_dataset_number/2.0)*2,:,:,:]
label_dataset_0 = label_dataset[:int(training_dataset_number/2.0),:,:,:]
label_dataset_1 = label_dataset[int(training_dataset_number/2.0)+1:int(training_dataset_number/2.0)*2,:,:,:]
print ('having 2 dataset in total, each of them has', int(training_dataset_number/2.0),'samples')

sess = tf.Session(config=tf.ConfigProto(log_device_placement=log_device_placement))
init_op = tf.global_variables_initializer()
sess.run(init_op)
saver = tf.train.Saver()

epoch = 150
batch_size = 128
total_num = len(training_dataset)

t2_1 = datetime.datetime.now()
for i in range(epoch):	# here is no shuffle
	for j in range(num_per_epoch):
		batch_images_0 = training_dataset_0[j*batch_size : (j+1)*batch_size]
		batch_labels_0 = label_dataset_0[j*batch_size : (j+1)*batch_size]
		batch_images_1 = training_dataset_1[j*batch_size : (j+1)*batch_size]
		batch_labels_1 = label_dataset_1[j*batch_size : (j+1)*batch_size]
		_, err, err1, err2 = sess.run([solver, total_loss_sum, loss_0, loss_1], feed_dict={images_0: batch_images_0, labels_0: batch_labels_0, images_1: batch_images_1, labels_1: batch_labels_1})
`

Was I wrong or did I miss something?

**Comment by kirilg** [comment]
Is this different from https://github.com/tensorflow/serving/issues/507? I think you'll have better luck asking this on the main [TensorFlow repo](https://github.com/tensorflow/tensorflow) or are you seeing a problem at serving time?
